src/metax_api/permissions/permissions.py

```diff
@@ -200,4 +200,9 @@ def has_object_permission(self, request, view, obj):
         """
         For service users, always returns True, since it is assumed they know what they are doing.
         """
-        return True
+
+        has_perm = obj.user_has_access(request)
+
+        if not has_perm:
+            self.message = 'You are not permitted to access this resource.'
+        return has_perm
```

Comment:
Sorry, I did not realize this before, but now this method is actually identical to what is defined in class ``EndUserPermissions``. In which case, it would be best to move this logic in class ``MetaxAPIPermissions`` (which currently raises ``NotImplementedError`` for this method), and remove these methods from the inheriting classes.

Reply:
sounds reasonable. will do